fog/overseer/server/tests/inactive_outstanding_key_idle_has_key.rs

```diff
@@ -0,0 +1,208 @@
+    std::fs::create_dir(&ledger_db_path).expect("couldn't create dir");
+    LedgerDB::create(&ledger_db_path).unwrap();
+    let mut ledger = LedgerDB::open(&ledger_db_path).unwrap();
+
+    // Make origin block before starting Fog Ingest servers
+    let origin_txo = utils::random_output(&mut rng);
+    let origin_contents = BlockContents {
+        key_images: Default::default(),
+        outputs: origin_txo.clone(),
+    };
+    let origin_block = Block::new_origin_block(&origin_txo);
+    ledger
+        .append_block(&origin_block, &origin_contents, None)
+        .expect("failed writing initial transactions");
+
+    let peer_indices = vec![0u16, 1u16, 2u16];
+    let (node0, state_file_node_0, client_listen_uri0) = utils::make_node(
+        0,
+        peer_indices.iter().cloned(),
+        recovery_db.clone(),
+        &watcher_path,
+        &ledger_db_path,
+        logger.clone(),
+    );
+    let (node1, _state_file_node_1, client_listen_uri1) = utils::make_node(
+        1,
+        peer_indices.iter().cloned(),
+        recovery_db.clone(),
+        &watcher_path,
+        &ledger_db_path,
+        logger.clone(),
+    );
+    let (node2, _state_file_node2, client_listen_uri2) = utils::make_node(
+        2,
+        peer_indices.iter().cloned(),
+        recovery_db.clone(),
+        &watcher_path,
+        &ledger_db_path,
+        logger.clone(),
+    );
+
+    assert!(!node0.is_active());
+    assert!(!node1.is_active());
+    assert!(!node2.is_active());
+
+    // Give RPC etc. time to start
+    std::thread::sleep(Duration::from_secs(1));
+
+    node0.activate().expect("node0 failed to activate");
+
+    assert!(node0.is_active());
+    assert!(!node1.is_active());
+    assert!(!node2.is_active());
+
+    let first_node0_ingress_key = node0.get_ingest_summary().get_ingress_pubkey().clone();
+    let first_node1_ingress_key = node1.get_ingest_summary().get_ingress_pubkey().clone();
+    let first_node2_ingress_key = node2.get_ingest_summary().get_ingress_pubkey().clone();
+
+    // All the keys should be equal.
+    assert_eq!(first_node0_ingress_key, first_node1_ingress_key);
+    assert_eq!(first_node0_ingress_key, first_node2_ingress_key);
+    assert_eq!(first_node1_ingress_key, first_node2_ingress_key);
+
+    // Initialize an OverSeerService object
+    let mut overseer_service = OverseerService::new(
+        vec![client_listen_uri0, client_listen_uri1, client_listen_uri2],
+        recovery_db.clone(),
+        logger.clone(),
+    );
+    overseer_service.start().unwrap();
+
+    // Set up the Rocket instance
+    let overseer_state = OverseerState { overseer_service };
+    // TODO: Consider testng the CLI here instead
+    let rocket_config: rocket::Config =
+        rocket::Config::build(rocket::config::Environment::Development)
+            // TODO: Make these either passed from CLI or in a Rocket.toml.
+            .address("127.0.0.1")
+            .port(80)
```

Comment:
Using the same port in multiple tests can have undesirable effects such as one of the tests failing to bind the port, or the requests going to the wrong place.

Like in the previous comment, I am not sure there's a need to bother with rocket here at all.

Reply:
See above.